src/sail_lean_backend/sail_plugin_lean.ml

```diff
@@ -145,10 +147,56 @@ let lean_rewrites =
     ("attach_effects", []);
   ]
 
-let output files = failwith "Lean backend not yet implemented"
+let create_lake_project_old (out_name : string) =
+  (* Change the directory if the option '--lean-output-dir' is set *)
+  let _ = match !opt_lean_output_dir with Some dir -> Sys.chdir dir | None -> () in
+  (* Call lake to create a new Lean project *)
+  match Sys.command ("lake new " ^ out_name) with
+  | 0 ->
+      Sys.chdir out_name;
+      (* Add the generated 'z3_problems' file to '.gitgnore' *)
+      let gitignore = open_out_gen [Open_creat; Open_text; Open_append] 0o640 ".gitignore" in
+      output_string gitignore "z3_problems\n";
+      close_out gitignore
+  | _ -> failwith "Failed to run lake command. Install Lean and lake via elan: https://github.com/leanprover/elan"
+
+let create_lake_project (out_name : string) =
+  (* Change the directory if the option '--lean-output-dir' is set *)
+  let _ = match !opt_lean_output_dir with Some dir -> Sys.chdir dir | None -> () in
+  (* Create a new directory if it doesn't exist already *)
+  Sys.mkdir out_name 0o775;
+  (* Switch to the new project directory *)
+  Sys.chdir out_name;
```

Comment:
It might be best to avoid using `chdir`, as we create our z3 cache (the `z3_problems` file) in the current working directory when we exit. If you `chdir` between when you start and exit, it won't find it.